include/swift/Driver/Compilation.h

```diff
@@ -258,6 +262,22 @@ class Compilation {
     }
   }
 
+  /// Returns whether the callee is permitted to pass -emit-tbd to a frontend
+  /// job.
+  ///
+  /// This only returns true once, because only one job should pass that
+  /// argument.
+  bool requestPermissionForFrontendToEmitTBD() {
+    if (PassedEmitTBDToFrontendJob)
+      // Someone else has already done it!
+      return false;
```

Comment:
Nitpick: please put braces around multi-line if statements even if the other line is just a comment.